src/services/security/modules/RedefineModule/index.ts

```diff
@@ -0,0 +1,98 @@
+import { REDEFINE_API_KEY } from '@/config/constants'
+import { type SafeTransaction } from '@safe-global/safe-core-sdk-types'
+import { generateTypedData } from '@safe-global/safe-core-sdk-utils'
+import { type SecurityResponse, type SecurityModule, SecuritySeverity } from '../types'
+
+const REDEFINE_URL = 'https://api.redefine.net/v2/risk-analysis/messages'
+
+export type RedefineModuleRequest = {
+  chainId: number
+  safeAddress: string
+  walletAddress: string
+  safeTransaction: SafeTransaction
+  threshold: number
+}
+
+export type RedefinedModuleResponse = RedefineResponse['data']['insights']
+
+type RedefinePayload = {
+  chainId: number
+  payload: {
+    method: 'eth_signTypedData_v4'
+    params: [string, string]
+  }
+}
+
+type RedefineSeverity = {
+  code: number
+  label: 'CRITICAL' | 'LOW' | 'NO_ISSUES'
+}
+
+type RedefineResponse = {
```

Comment:
They don't provide official types?

Reply:
No. Not yet at least. I asked for a sdk / typescript types ~2 weeks ago.